checkapi/main.go

```diff
@@ -165,24 +167,31 @@ func walkFolder(cfg internal.Config, folder string, componentType string) error
 		return errors.Join(errs...)
 	}
 
-	if _, err := os.Stat(filepath.Join(folder, "config.schema.json")); errors.Is(err, os.ErrNotExist) {
+	if metadata.Config == nil {
 		if cfg.JSONSchema.CheckPresent {
 			errs = append(errs, err)
 		}
 	} else {
-		jsonSchema, err := internal.ReadJSONSchema(folder)
+		schemaBytes, err := json.Marshal(metadata.Config)
 		if err != nil {
 			errs = append(errs, err)
 		} else {
-			var after *jsonschema.Schema
-			if after, err = internal.DeriveSchema(*cfgStruct, result.Structs, cfg.JSONSchema.TypeMappings); err != nil {
+			jsonSchema, err := jsonschema.NewCompiler().Compile(schemaBytes)
+			if err != nil {
 				errs = append(errs, err)
-			} else if err := internal.CompareJSONSchema(folder, jsonSchema, after); err != nil {
-				errs = append(errs, err)
-				b, _ := after.MarshalJSON()
-				errs = append(errs, fmt.Errorf("[%s] new JSON schema: %s", folder, string(b)))
+			} else {
+				var after *jsonschema.Schema
```

Comment:
`after` is meaningless to me in this context, `structDerivedSchema` or simply `derivedSchema` perhaps?